components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlProducer.java

```diff
@@ -277,6 +277,8 @@ protected void processStreamList(Exchange exchange, PreparedStatementCreator sta
             }
 
             boolean isResultSet = ps.execute();
+            //pass through all headers
+            exchange.getOut().getHeaders().putAll(exchange.getIn().getHeaders());
```

Comment:
Can you move this code inside the if (isResultSet) so the headers is only pass through if we set some data because there was a result set.